### ISSUE TITLE
React Signup and Forgot Password Links not Hidden on SignIn Page Component

### ISSUE THREAD

**Issue by nicklee1990** [aws-amplify/amplify-js#1599]
**Describe the bug**
When using the default SignIn component provided by the `aws-amplify-react` package, if the SignUp and ForgotPassword components are omitted from the `authenticatorComponents` argument to withAuthenticator, or omitted as children of an `Authenticator`, the links should on the SignIn page should be hidden (L121 and L122 - set unused variables). Clicking these links causes an error when they are omitted.

If agreed this is a bug I'll submit a PR to fix.

**To Reproduce**
1. Setup a basic React App that uses the `withAuthenticator` HOC
2. Set the 3rd argument to withAuthenticator to: 
    ```
    [
      <SignIn/>,
      <ConfirmSignIn/>,
      <Greetings/>,
    ]
    ```
3. Load the Sign In page in your browser

**Expected behavior**
The Sign up and Forgot Password  links should be hidden when SignUp and/or ForgotPassword components are **not** passed to the Authenticator

**Additional context**
Looks like it's caused by this L157 and L172 in the SignIn Component not using the `hideForgotPassword` and `hideSignUp` vars defined earlier in the `showComponent` method
https://github.com/aws-amplify/amplify-js/blob/master/packages/aws-amplify-react/src/Auth/SignIn.jsx#L172

i.e.

```
{
    hideSignUp && <SectionFooterSecondaryContent theme={theme}>
        {I18n.get('No account? ')}
        <Link theme={theme} onClick={() => this.changeState('signUp')}>
            {I18n.get('Create account')}
        </Link>
    </SectionFooterSecondaryContent>
}
```

**Comment by mogadanez** [comment]
it was  there  and reverted back again

https://github.com/aws-amplify/amplify-js/blame/e1507586f5e654420822841dc2ec22014402c61f/packages/aws-amplify-react/src/Auth/SignIn.jsx

https://github.com/aws-amplify/amplify-js/blame/master/packages/aws-amplify-react/src/Auth/SignIn.jsx#L172

**Comment by powerful23** [comment]
@nicklee1990 Hey this has been fixed in #1632

**Comment by nicklee1990** [comment]
Nice!
